pfuture.el

```diff
@@ -27,21 +27,71 @@
 (require 'cl-lib)
 (require 'inline)
 
-;;;###autoload
-(defun pfuture-new (cmd &rest cmd-args)
-  "Create a new future process for command CMD and arguments CMD-ARGS.
-This will return a process object with one additional 'result property which
-can be read via \(process-get process 'result\) or alternatively with
-\(pfuture-result process\).
+(defvar pfuture--dummy-buffer nil
+  "Dummy buffer for stderr pipes.")
+
+(define-inline pfuture--delete-process (process)
+  "Delete PROCESS with redisplay inhibited."
+  (inline-letevals (process)
+    (inline-quote
+     (let ((inhibit-redisplay t))
+       (delete-process ,process)))))
+
+(defun pfuture--sentinel (process _)
+  "Delete the stderr pipe process of PROCESS."
+  (unless (process-live-p process)
+    (let* ((stderr-process (process-get process 'stderr-process)))
+      ;; Set stderr-process to nil so that await-to-finish does not delete
+      ;; the buffer again.
+      (process-put process 'stderr-process nil)
+      ;; delete-process may trigger other sentinels. If there are many pfutures,
+      ;; this will overflow the stack.
+      (run-with-idle-timer 0 nil #'pfuture--delete-process stderr-process))))
 
-Note that CMD-ARGS must be a *sequence* of strings, meaning
+;;;###autoload
+(defun pfuture-new (&rest cmd)
+  "Create a new future process for command CMD.
+Any arguments after the command are interpreted as arguments to the command.
+This will return a process object with additional 'stderr and 'stdout
+properties, which can be read via \(process-get process 'stdout\) and
+\(process-get process 'stderr\) or alternatively with
+\(pfuture-result process\) or \(pfuture-stderr process\).
+
+Note that CMD must be a *sequence* of strings, meaning
 this is wrong: (pfuture-new \"git status\")
 this is right: (pfuture-new \"git\" \"status\")"
-  (let* ((process-connection-type nil)
-         (process (apply #'start-process "Process Future" nil cmd cmd-args)))
-    (process-put process 'result "")
-    (set-process-filter process #'pfuture--append-output)
-    process))
+  (let ((stderr (make-pipe-process
```

Comment:
Creating the pipe process separately instead of using a buffer avoids problems with buffer hooks and redisplay.